### DISCUSSION TITLE
TTGO T8 V1.1

### DISCUSSION THREAD

**Discussion by scribblrsam** [HASwitchPlate/openHASP#198] [Hardware]
Is it possible to use this board with openHASP? 

I want to try openHASP and can get this board and an MSP touch screen delivered pretty quickly. Looking at the pins required for the MSP display [here](https://haswitchplate.github.io/openHASP-docs/0.6.1/displays/MSPxxxx/) and the pinout [here](https://github.com/LilyGO/TTGO-T8-ESP32) it seems that the board doesn't have  pin 17 available. 

I assume it is possible to override this and have seen the GPIO template in`user_config_override-template.h` but cant find any docs about how this works. 

Thanks

**Comment by fvanroie** [comment]
The default pin configuration is based on the `user_setups\esp32\d1-mini-esp32_ili9341` build.
Indeed, on dev boards with PSram `GPIO16` and `GPIO17` are not available because they are used by the PSram chip.

You can _most likely_ use the [`lolin-d32-pro_ili9341`](https://github.com/HASwitchPlate/openHASP/blob/34112882a17c1400968fe794df65d4fdac8b31a8/user_setups/esp32/lolin-d32-pro_ili9341.ini#L28-L32) configuration, since that board also has PSram. But the TTGO only has 4MB flash... that needs to be adjusted in the ini file too.

It defines the same SPI pins but uses 5 different control pins:
```ini
    ;-D TFT_DC=27   ; Defined by board, don't redefine !!
    ;-D TFT_CS=14   ; Defined by board, don't redefine !!
    ;-D TFT_RST=33  ; Defined by board, don't redefine !!
    -D TFT_BCKL=32  ; Configurable via web UI (default 32)
    -D TOUCH_CS=12  ; Default for TFT connector
```
So replace these 5 GPIO's in the ESP32 Pin column and you should be good to go!

> Note: Leave `SDO(MISO)` of the display unconnected!

To build this configuration, you can use [Visual Studio Code](https://haswitchplate.github.io/openHASP-docs/0.6.1/compiling/local/#create-a-configuration) or [Gitpod](https://haswitchplate.github.io/openHASP-docs/0.6.1/compiling/gitpod/#choose-the-version-you-need). In both cases you need to copy and edit the `platformio_override.ini` file from the template to enable the ESP32 environment(s) you want.

**Reply by fvanroie** [reply]
The custom configuration might look something like this in platform_override.ini:

Add the `my_custom_build` to `extra_default_envs `:

```ini
;region -- Default Build Environments : Used when Build All ---
extra_default_envs =
    ; Uncomment specific environments or create extra:
    my_custom_build
;endregion
```

Then replace the `[env:my_custom_build]` section with:

```ini
;region -- Custom Environment configuration example -----------------
[env:my_custom_build]
extends = esp32
board = esp32dev

build_flags =
    ${env.build_flags}
    ${esp32.build_flags}
    ${esp32.ps_ram}
    -D HASP_MODEL="Custom TTGO T8 ILI9341"

;region -- TFT_eSPI build options ------------------------
    ${lcd.lolin24}
    ${esp32.vspi}        ; Use VSPI hardware SPI bus
    -D USE_TFT_ESPI
    -D TFT_DC=27  
    -D TFT_CS=14  
    -D TFT_RST=33 
    -D TFT_BCKL=32  ; Configurable via web UI (default 32)
    -D TOUCH_CS=12
;endregion

;region -- Library options -------------------------------
lib_deps =
    ${env.lib_deps}
    ${esp32.lib_deps}

lib_ignore =
    ${env.lib_ignore}
    ${esp32.lib_ignore}
;endregion
```

**Reply by scribblrsam** [reply]
@fvanroie - That's brilliant!! Thanks for the quick and detailed response.

**Reply by nagyrobi** [reply]
Please confirm if it's working for you or not, we might add it to the compatibility list.

**Reply by scribblrsam** [reply]
I ended up getting an M5Stack to test. Will try this soon hopefully and let you know.
